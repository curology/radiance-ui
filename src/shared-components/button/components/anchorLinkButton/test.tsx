import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { AnchorLinkButton } from './index';

describe('<AnchorLinkButton/>', () => {
  describe('UI snapshots to double-check regressions', () => {
    it('renders with props', () => {
      const { container } = render(
        <AnchorLinkButton onClick={() => undefined}>Click me</AnchorLinkButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('passes down all valid button props to underlying button', () => {
      const label = 'Click me for more information';

      const { getByLabelText } = render(
        <AnchorLinkButton
          onClick={() => undefined}
          aria-label={label}
        >
          Click me
        </AnchorLinkButton>,
      );

      const button = getByLabelText(label);
      expect(button).toHaveTextContent('Click me');
    });
  });
});
