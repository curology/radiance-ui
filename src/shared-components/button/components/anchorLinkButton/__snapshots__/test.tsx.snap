// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnchorLinkButton/> UI snapshots to double-check regressions renders with props 1`] = `
.emotion-0 {
  border-bottom: 1px solid currentColor;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  background: transparent;
  border: none;
  padding: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-skip-ink: none;
  font-size: 0.875rem;
  border-radius: 4px;
}

.emotion-0:hover {
  opacity: 0.6;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

<button
  class="emotion-0"
  type="button"
>
  Click me
</button>
`;
